src/Units/UnitState.java

```diff
@@ -87,64 +87,62 @@ public boolean hasMaterials()
 
   public boolean isHurt()
   {
-    return health < healthFromHP(UnitModel.MAXIMUM_HP);
+    return getHP() < UnitModel.MAXIMUM_HP;
   }
   public int getHP()
   {
-    return (int) Math.ceil(healthToHP(health));
+    return roundHP(health);
   }
-  /** @return value in range [0-1.0]; represents the unit's current effectiveness */
-  public double getHPFactor()
+  public int getHPFactor()
   {
-    return getHP() / (double) UnitModel.MAXIMUM_HP;
+    return roundHP(health) / 10;
   }
-  /** @return un-rounded HP */
-  public double getPreciseHP()
+  public static int roundHP(int health)
   {
-    return healthToHP(health);
-  }
-  public static double healthToHP(int input)
-  {
-    return ((double) input) / 10;
-  }
-  public static int healthFromHP(double input)
-  {
-    return (int) (input * 10);
+    if( health >= 0 )
+      //     "round up", then kill the last digit
+      return (health + 9) / 10 * 10;
+    // Truncation rounds toward zero, so we need to round down for negative values
+    return (health - 9) / 10 * 10;
   }
 
   /**
    * Reduces HP by the specified amount.
    * <p>Enforces a minimum (optional) of 0.
    * <p>Use this for lethal damage, especially unit-on-unit violence. Do not use for healing.
-   * @return the change in HP
+   * @return the change in *rounded* HP
    */
-  public int damageHP(double damage)
+  public int damageHP(int damage)
   {
     return damageHP(damage, false);
   }
-  public int damageHP(double damage, boolean allowOverkill)
+  public int damageHP(int damage, boolean allowOverkill)
   {
     if( damage < 0 )
       throw new ArithmeticException("Cannot inflict negative damage!");
     int before = getHP();
-    health = health - healthFromHP(damage);
+    health = health - damage;
     if( !allowOverkill )
       health = Math.max(0, health);
     return getHP() - before;
   }
 
   /**
    * Increases HP by the specified amount.
-   * <p>Enforces a minimum of 0.1, and a maximum (optional) of MAXIMUM_HP.
-   * <p>When healing, rounds health up to a whole HP (e.g. 2.5 + 2 = 4.5 -> 5.0)
+   * <p>Enforces a minimum of 1, and a maximum (optional) of MAXIMUM_HP.
+   * <p>When healing, rounds health up to a whole HP (e.g. 25 + 20 = 45 -> 50)
```

Comment:
Is there anywhere we define the relationship between health and HP? If I didn't already know what I was looking at, this comment could be pretty confusing.

Reply:
Ah, back on line 27-28 I guess, where `health` is defined.

Reply:
They're uh... not really different anymore? I guess I should say "GUI HP" or something?
I dunno. Nomenclature is hard

Reply:
Oh. Then the aforementioned comment should probably be updated?